src/plugins/charts/public/static/components/color_picker.tsx

```diff
@@ -74,54 +74,87 @@ export const legendColors: string[] = [
 ];
 
 interface ColorPickerProps {
+  /**
+   * Optionally set color picker id.
+   */
   id?: string;
+  /**
+   * Label that characterizes the color that is going to change
+   */
   label: string | number | null;
+  /**
+   * Callback on the color change
+   */
   onChange: (color: string | null, event: BaseSyntheticEvent) => void;
+  /**
+   * Initial color.
+   */
   color: string;
+  /**
+   * Defines if the compatibility (legacy) or eui palette is going to be used. Defauls to true.
+   */
+  useLegacyColors?: boolean;
+  /**
+   * Defines if the default color is overwritten. Defaults to true.
+   */
+  colorIsOverwritten?: boolean;
 }
 
-export const ColorPicker = ({ onChange, color: selectedColor, id, label }: ColorPickerProps) => (
-  <div className="visColorPicker">
-    <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
-      <FormattedMessage
-        id="charts.colorPicker.setColor.screenReaderDescription"
-        defaultMessage="Set color for value {legendDataLabel}"
-        values={{ legendDataLabel: label }}
-      />
-    </span>
-    <div className="visColorPicker__value" role="listbox">
-      {legendColors.map((color) => (
-        <EuiIcon
-          role="option"
-          tabIndex={0}
-          type="dot"
-          size="l"
-          color={selectedColor}
-          key={color}
-          aria-label={color}
-          aria-describedby={`${id}ColorPickerDesc`}
-          aria-selected={color === selectedColor}
-          onClick={(e) => onChange(color, e)}
-          onKeyPress={(e) => onChange(color, e)}
-          className={classNames('visColorPicker__valueDot', {
-            // eslint-disable-next-line @typescript-eslint/naming-convention
-            'visColorPicker__valueDot-isSelected': color === selectedColor,
-          })}
-          style={{ color }}
-          data-test-subj={`visColorPickerColor-${color}`}
+const euiColors = euiPaletteColorBlind({ rotations: 4, order: 'group' });
+
+export const ColorPicker = ({
+  onChange,
+  color: selectedColor,
+  id,
+  label,
+  useLegacyColors = true,
+  colorIsOverwritten = true,
+}: ColorPickerProps) => {
+  const legendColors = useLegacyColors ? legacyColors : euiColors;
+
+  return (
+    <div className="visColorPicker">
+      <span id={`${id}ColorPickerDesc`} className="euiScreenReaderOnly">
+        <FormattedMessage
+          id="charts.colorPicker.setColor.screenReaderDescription"
+          defaultMessage="Set color for value {legendDataLabel}"
+          values={{ legendDataLabel: label }}
         />
-      ))}
+      </span>
+      <div className="visColorPicker__value" role="listbox">
```

Comment:
I see that this is only moved but can we please improve this?

[First rule of aria](https://www.w3.org/TR/using-aria/#firstrule), don't use aria. Why is it the first rule? Because when we write custom semantics, they're hard to get right. And here, with `role=listbox` we declare we're going to have the same semantics as an HTML `<select>` but then we don't follow through on that promise as we don't implement all the necessary keyboard interactions on the options below. 

So let's simplify the semantics. 

```suggestion
      <fieldset className="visColorPicker__value">
      	<EuiScreenReaderOnly><legend>
      		<FormattedMessage
          		id="charts.colorPicker.setColor.screenReaderDescription"
          		defaultMessage="Set color for value {legendDataLabel}"
          		values={{ legendDataLabel: label }}>
          	/>
         </legend></EuiScreenReaderOnly>
```

Here I moved the faux-label created above into the `fieldset` and then into a `legend`. This sets up a defined area for some related inputs and a name for all of them. I also moved from using the classname `.euiScreenReaderOnly` to using the actual component as all of the classnames are implementation details and are not guaranteed to keep working whereas the components are the public API of EUI.

Setting up this `legend` also lets us get rid of the `id` value that current'y isn't working for some reason and is being rendered as `undefinedColorPickerDesc` which, I guess, is at least consistent but is not the intention of how this should work.